### ISSUE TITLE
Move the take over mode to the input implementation

### ISSUE THREAD

**Issue by rdner** [elastic/beats#36777]
**Describe the enhancement:**

Currently, the take-over mode used for migrating log inputs to filestream inputs is running before we start any inputs and migrates everything at once for the best possible performance.

We might be able to move this logic to the filestream's [prospector](https://github.com/elastic/beats/blob/7e61ff00bbbb873567c809c42f6a4cd5f31617e6/filebeat/input/filestream/prospector.go#L131) when it starts instead. The prospector has access to the state via this interface:

https://github.com/elastic/beats/blob/7e61ff00bbbb873567c809c42f6a4cd5f31617e6/filebeat/input/filestream/internal/input-logfile/prospector.go#L39-L49

We might want to extend the interface and add the take over function.

This will allow us to support take-over for dynamic use-cases but would most likely affect the performance.

**Describe a specific use case for the enhancement or feature:**

If we allow dynamic take-over it will allow us to use it in autodiscover and running under elastic-agent.

**Comment by rdner** [comment]
I'm prioritizing this because it's a blocker for migrating integration packages to using the `filestream` input.

**Comment by rdner** [comment]
Closing in favor of https://github.com/elastic/beats/issues/42884
